staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -551,14 +551,18 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		if err != nil {
 			return nil, nil, err
 		}
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			return nil, nil, err
+		}
 
 		// If AllowUnconditionalUpdate() is true and the object specified by
 		// the user does not have a resource version, then we populate it with
 		// the latest version. Else, we check that the version specified by
 		// the user matches the version of latest storage object.
 		resourceVersion, err := e.Storage.Versioner().ObjectResourceVersion(obj)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, invalidResourceVersion(qualifiedResource, name, accessor.GetResourceVersion(), fmt.Sprintf("Failed to parse resource version: %v", err))
```

Comment:
thinking through this further, since the resourceVersion is intended to be opaque to the API consumer, saying "failed to parse resourceVersion" and omitting the underlying error is probably better... we don't really want them trying to react to the error message by tweaking the data they send, we want them echoing a resourceVersion we sent them back